src/OrbitGl/TimelineTicks.h

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_GL_TIMELINE_IMPL_H_
+#define ORBIT_GL_TIMELINE_IMPL_H_
+
+#include <cstdint>
+#include <set>
+#include <vector>
+
+namespace orbit_gl {
+
+constexpr uint64_t kSecondToNano = 1000 * 1000 * 1000;
+constexpr uint64_t kMinuteToNano = 60 * kSecondToNano;
+constexpr uint64_t kHourToNano = 60 * kMinuteToNano;
+
+class TimelineTicks {
+  enum class TickType { MajorTick, MinorTick };
+
+ public:
+  TimelineTicks();
+
+  // For now, only major ticks.
+  [[nodiscard]] std::vector<std::pair<TickType, uint64_t> > GetTicks(uint64_t start_ns,
+                                                                     uint64_t end_ns) const;
+  [[nodiscard]] std::vector<uint64_t> GetMajorTicks(uint64_t start_ns, uint64_t end_ns) const;
+
+ private:
+  uint64_t GetScale(uint64_t time_range_ns) const;
+  std::set<uint64_t> scales_;
+};
```

Comment:
I'm not sure why this needs to be a class. `scales_` seems to be a member that never changes. How you consider making it a constant defined in the cpp file and just have free functions here? Or maybe keep the class and just move `scales_` out of it?

Or is there maybe more coming?

Reply:
Why do you need a class to unit-test it?

Und just to emphasize: Keeping the class is okay, but having `scales_` as member that both never changes and is not a constructor parameter feels wrong. You could just make it a constant and be good with it.

Reply:
It's done!